src/org/folio/models/FolioInstallJson.groovy

```diff
@@ -1,102 +1,217 @@
   }
 
+  /**
+   * Adds multiple modules with their respective actions to the installJsonObject.
+   *
+   * @param modules a list of maps where each map contains the module ID and its action.
+   */
   void addModulesWithActions(List<Map<String, String>> modules) {
-    modules.each { module ->
-      addModule(module['id'], module['action'])
-    }
+    modules.each { module -> addModule(module['id'], module['action']) }
   }
 
+  /**
+   * Adds multiple modules with the same action to the installJsonObject.
+   *
+   * @param moduleIds a list of module IDs to be added.
+   * @param action the action to perform on all specified modules.
+   */
   void addModulesWithSameAction(List<String> moduleIds, String action) {
-    moduleIds.each { moduleId ->
-      addModule(moduleId, action)
-    }
+    moduleIds.each { moduleId -> addModule(moduleId, action) }
   }
 
+  /**
+   * Removes a module from the installJsonObject by its name.
+   *
+   * @param name the name of the module to remove.
+   */
   void removeModuleByName(String name) {
-    this.installJsonObject.removeAll { module ->
-      module.name == name
-    }
+    this.installJsonObject.removeAll { module -> module.name == name }
   }
 
+  /**
+   * Removes multiple modules from the installJsonObject by their names.
+   *
+   * @param names a list of names corresponding to the modules to remove.
+   */
   void removeModulesByName(List<String> names) {
-    names.each { name ->
-      removeModuleByName(name)
-    }
+    names.each { name -> removeModuleByName(name) }
   }
 
-  List getBackendModules() {
+  /**
+   * Retrieves all backend modules from the installJsonObject.
+   *
+   * @return a list of backend FolioModules.
+   */
+  List<FolioModule> getBackendModules() {
     return _getModulesByType(ModuleType.BACKEND)
   }
 
-  List getEdgeModules() {
+  /**
+   * Retrieves all edge modules from the installJsonObject.
+   *
+   * @return a list of edge FolioModules.
+   */
+  List<FolioModule> getEdgeModules() {
     return _getModulesByType(ModuleType.EDGE)
   }
 
-  List getUiModules() {
+  /**
+   * Retrieves all UI modules from the installJsonObject.
+   *
+   * @return a list of frontend FolioModules.
+   */
+  List<FolioModule> getUiModules() {
     return _getModulesByType(ModuleType.FRONTEND)
   }
 
+  /**
+   * Retrieves the Okapi module from the installJsonObject.
+   *
+   * @return the FolioModule representing Okapi, or null if not found.
+   */
   FolioModule getOkapiModule() {
-    return this.installJsonObject.find { module ->
-      module.getType() == ModuleType.OKAPI
-    }
+    return this.installJsonObject.find { module -> module.getType() == ModuleType.OKAPI }
   }
 
-  List getInstallJson() {
+  /**
+   * Converts the current installJsonObject into a list of maps suitable for installation JSON.
+   *
+   * @return a list of maps containing module IDs and their actions.
+   */
+  List<Map<String, String>> getInstallJson() {
```

Comment:
Why to use nested objects? It will require double encapsulation, double RAM, slow Jenkins...

Reply:
This is compromise to reduce code duplicates and complex logic in different flows